### PR TITLE
Allow overriding the classifier in case no main artifact exists

### LINKED ISSUE #39: Check if artifact exists in remote repository not working for test-jar
Hi

First of all, I really like the idea behind the plugin and I have use cases where it is really helpful.

However, I have a problem with the following constellation:

We have Maven modules which expose code from the test scope (but they do not produce any main artifacts) by creating `test-jar` artifacts to use them in other modules, referenced with the `type` element:

```
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-jar-plugin</artifactId>
    <executions>
      <execution>
        <goals>
          <goal>test-jar</goal>
        </goals>
      </execution>
    </executions>
</plugin>
```

Let's call the artifact 'testArtifact'

In the Maven module where these test jars are used, let's call it `testApplication`, they are declared as dependencies:

```
<dependency>
  <groupId>some.groupId</groupId>
  <artifactId>testArtifact</artifactId>
  <version>${project.version}</version>
  <type>test-jar</type>
</dependency>
```

Because they need to be packaged into the resulting jar.

the `exists-maven-plugin` correctly identifies that the `testApplication` artifact already exists in the remote repository. However, because the deployment of the primary artifact is skipped (deploy.skip=true), Maven has some fallback behavior which results in this:

```
[exists-maven-plugin] some.groupId:testApplication:jar:version exists
[exists-maven-plugin] setting maven.deploy.skip=true
[exists-maven-plugin] Skipping artifact deployment
[maven] No primary artifact to deploy, deploying attached artifacts instead.
[maven] Uploading to ***-repository: https://nexus/content/repositories/release-repository/groupid/testArtifact/version/testArtifact.artifactId-version.pom
```

And then the build fails because it tries to deploy an artifact that already exists.

Do you have a suggestion how I can solve this problem? Or is this a bug?

### PR DESCRIPTION
**Pull request by tomandegg**
fixes #39 
- I decided to try a simple solution first with an additional configuration element `<classifier>`
- Adding the classifier to `<project>` isn't straight-forward, as `${packaging}` currently is optional (even though it is not declared so in the README). So adding the classifier there as another optional element would need some logic to determine which optional group is present - doable, but I didn’t know if you’d like the ‘complexity’. Another option would be to add the classifier as a mandatory group, but that would be a breaking change.
- I first wanted to get your feedback about this solution or whether you’d rather go another route. If we stay with this, the logging in {{AbstractExistsMojo.java}} (line 200 et. al.) could be adapted to display the classifier in addition to just the {{project}}, but it would need some changes to common code, so I wanted to postpone this work until after your feedback, if at all, because the classifier is visible in the artifact location that is checked and logged, so it might not give much more insight, after all.
- I had to bump the version of `maven-install-plugin`for the integration test as without the rather new `<allowIncompleteProjects>true</allowIncompleteProjects>` option, I wasn’t able to make the new integration tests work.

### CODE REVIEW

File: src/main/java/org/honton/chas/exists/GAV.java [hunk 1]
```diff
@@ -12,9 +12,10 @@ public class GAV {
   final String groupId;
   final String artifactId;
   final String type;
+  final String classifier;
   final String version;
 
-  GAV(String project, String packaging) throws MojoFailureException {
+  GAV(String project, String packaging, String configuredClassifier) throws MojoFailureException {
```
Comment (chonton) [inline src/main/java/org/honton/chas/exists/GAV.java:18]: flip order of second and third parameters.

**Comment by chonton** [comment]
Looks pretty good.  Thanks for the tests.
You are welcome to add yourself to the `<developers>`

**Comment by tomandegg** [comment]
Thanks for your feedback!

Today I finally found some time to test the changes in our production project and can confirm that this PR fixes the issue.

One further thought: While working on this feature, I got to the conclusion that if it were my project, I would probably have implemented this in a different way. 
- adapt the `project` configuration such that it includes the `classifier`
- change the implementation about handling the `packaging` such that it would be mandatory, with the reasoning that anybody who uses this non-standard configuration probably knows quite well what she/he is doing
- and maybe even straighten out some inconsistencies concerning terminology where sometimes `type` and `packaging` are used interchangeably, and adhere more to the matrix [here](https://maven.apache.org/ref/3.9.1/maven-core/artifact-handlers.html) where the `type` is much more important than `packaging`, but is not synonymous

These are just some thoughts / ideas that I wanted to share. I did not want to put all of this into this very first PR as it would mean some larger refactoring and breaking changes. But in case these are thoughts that have been floating around in your head for some time, too, let me know. Otherwise, just disregard all except the first three lines of this comment and I'm absolutely fine if this change is merged as is.

**Comment by chonton** [comment]
I'll merge this work as is.  I am looking forward to additional pull requests!

**Comment by tomandegg** [comment]
Great to hear. Do you have plans to release a new version of the plugin to include this change?

Also, am I interpreting your comment correctly: Would you welcome a PR which implements the conceptual outline I described in my last comment? I cannot promise when I'll find time for this, but if you agree conceptually, I could leverage a moment of motivation at some point to invest some time.

**Comment by chonton** [comment]
0.9.0 released
Yes, PR is welcome

**Pull request merged**